mypy/plugins/dataclasses.py

```diff
@@ -395,6 +398,17 @@ def transform(self) -> bool:
 
         return True
 
+    def _add_dunder_replace(self, attributes: list[DataclassAttribute]) -> None:
+        """Add a `__replace__` method to the class, which is used to replace attributes in the `copy` module."""
+        args = [attr.to_argument(self._cls.info, of="replace") for attr in attributes]
+        add_method_to_class(
+            self._api,
+            self._cls,
+            "__replace__",
+            args=args,
+            return_type=Instance(self._cls.info, []),
```

Comment:
This seems like it might be incorrect for a generic dataclass. Can you add some tests involving generic dataclasses?

Reply:
Yeah, the actual `Self` type didn't work on `self._cls.self_type` (was just `None`), so wasn't 100% sure, but let me at least test that to see.